int32[] botIds
int32[] status